src/main/java/com/amazon/ion/impl/macro/Expression.kt

```diff
@@ -221,21 +237,25 @@ sealed interface Expression {
      */
     data class VariableRef(val signatureIndex: Int) : TemplateBodyExpression
 
+    sealed interface InvokableExpression : HasStartAndEnd, Expression {
```

Comment:
🗺️ We're able to reduce some code duplication by unifying the code paths for TDL and E-Expression macros. This interface helps to facilitate that unification.